### ISSUE TITLE
How FrozenBatchNorm2d load the pretrained model parameters

### ISSUE THREAD

**Issue by Huangdebo** [facebookresearch/detr#327]
Thanks for the code of DETR. 
When I read the cod [backbone](https://github.com/facebookresearch/detr/blob/master/models/backbone.py), I don't understand how does the  FrozenBatchNorm2d work:

1. The backbone uses the FrozenBatchNorm2d, so the key of BN in state_dict changes. Now when the resnet loads the pretrained model, how it loads the pretrained parameters?
2. The resnet uses load_state_dict to load whole pretrained model. But there is still a funciton _load_from_state_dict in FrozenBatchNorm2d, how does it works? 

Looking forward to your reply @lessw2020 @fmassa

**Comment by alcinos** [comment]
> so the key of BN in state_dict changes
Actually they don't, by design the frozen batch norm uses the same parameter names, precisely to make loading possible.
If you see the code, one key is removed ("num_batches_tracked") because it's not used in the frozen batch norm, and the others are loaded transparently because they have the same names.

I believe I have answered your question, and as such I'm closing this. Feel free to reach out if you have further concerns.

**Comment by Huangdebo** [comment]
“one key is removed ("num_batches_tracked")“
Sorry, i'm still confused. The resnet uses load_state_dict to load pretrained model, how can  _load_from_state_dict work to  removed the key ("num_batches_tracked")? i can't find where to call this function

**Comment by ma3252788** [comment]
I'm also wondering about FrozenBatchNorm2d
